openedx/core/djangoapps/content/block_structure/management/commands/generate_course_blocks.py

```diff
@@ -133,7 +133,7 @@ def _generate_course_blocks(self, options, course_keys):
         Generates course blocks for the given course_keys per the given options.
         """
         if options.get('with_storage'):
-            waffle().override_for_request(STORAGE_BACKING_FOR_CACHE)
+            waffle().set_request_cache_with_short_name(STORAGE_BACKING_FOR_CACHE, True)
```

Comment:
I'm sure we'll come back around to this, but let's just find another way to do this.  

If the options are not available in all the places that the switch is used, it would probably be as simple as storing OVERRIDE_STORAGE_BACKING_FOR_CACHE in a RequestCache, and having a function that first checks that value, and then falls back to the waffle switch value.  That function could be called in place of checking the switch elsewhere.

Reply:
I understand what you mean. Fortunately, all access to the `STORAGE_BACKING_FOR_CACHE` switch is performed via `block_structure.store._is_storage_backing_enabled`, so what you are suggesting is doable. Shall we do that in this PR or in another one?

Reply:
Follow-up PR.  Feel free to add a task.  I think you'll need this as part of Namespace retiring effort anyway.